src/openms/source/METADATA/ID/IdentificationDataConverter.cpp

```diff
@@ -1214,4 +1214,169 @@ namespace OpenMS
+        ID::ObservationMatch match(feature.getPrimaryID(), obs_ref,
+                                   feature.getCharge());
+        ID::ObservationMatchRef match_ref = id_data.registerObservationMatch(match);
+        feature.addIDMatch(match_ref);
+      }
+      for (ID::ObservationMatchRef ref : feature.getIDMatches())
+      {
+        // store trace of feature indexes so we can map the converted ID back;
+        // key needs to be unique in case the same ID matches multiple features:
+        String key = "IDConverter_trace_" + String(id_counter);
+        id_data.setMetaValue(ref, key, i);
+        ++id_counter;
+      }
+    }
+
+    exportIDs(consensus.getIdentificationData(), consensus.getProteinIdentifications(),
+              consensus.getUnassignedPeptideIdentifications(), false);
+
+    // map converted IDs back to features using meta values assigned above;
+    // in principle, different "observation matches" from one "observation"
+    // can map to different features, which makes things complicated when they
+    // are converted to "peptide hits"/"peptide identifications"...
+
+    auto& pep_ids = consensus.getUnassignedPeptideIdentifications();
+    for (Size i = 0; i < pep_ids.size(); )
+    {
+      PeptideIdentification& pep = pep_ids[i];
+      // move hits outside of peptide ID so ID can be copied without the hits:
+      vector<PeptideHit> all_hits;
+      all_hits.swap(pep.getHits());
+      vector<bool> assigned_hits(all_hits.size(), false);
+      // which hits map to which features:
+      map<ConsensusFeature*, set<Size>> features_to_hits;
+      for (Size j = 0; j < all_hits.size(); ++j)
+      {
+        PeptideHit& hit = all_hits[j];
+        vector<String> meta_keys;
+        hit.getKeys(meta_keys);
+        for (const String& key : meta_keys)
+        { // ID-data stores a trace (feature index) which is used for a lookup
+          // to attach the converted IDs back to the specific feature.
+          if (key.hasPrefix("IDConverter_trace_"))
+          {
+            Size index = hit.getMetaValue(key);
+            hit.removeMetaValue(key);
+            ConsensusFeature* feat_ptr = &consensus.at(index);
+            features_to_hits[feat_ptr].insert(j);
+            assigned_hits[j] = true;
+          }
+        }
+      }
+      // copy peptide ID with corresponding hits to relevant features:
+      for (auto& pair : features_to_hits)
+      {
+        auto& feat_ids = pair.first->getPeptideIdentifications();
+        feat_ids.push_back(pep);
+        for (Size hit_index : pair.second)
+        {
+          feat_ids.back().getHits().push_back(all_hits[hit_index]);
+        }
+      }
+
+      bool all_assigned = all_of(assigned_hits.begin(), assigned_hits.end(),
+                                 [](bool b) { return b; });
+      if (all_assigned) // remove peptide ID from unassigned IDs
+      {
+        pep_ids.erase(pep_ids.begin() + i);
```

Comment:
this looks a bit fishy... could this be a major bottleneck (e.g., let's say we have a >50gb consensusXML with many ids as regularly with quantMS)?

Reply:
Possibly. Needs testing (obviously).

Reply:
Can you test it? Otherwise, this could easily be one of these things that will break an analysis and requires more work to fix later...

Reply:
a sizable testfile is available here: http://ftp.pride.ebi.ac.uk/pub/databases/pride/resources/proteomes/absolute-expression/PXD000561/proteomicslfq/